addons/sale/models/payment.py

```diff
@@ -63,32 +63,29 @@ def _set_transaction_pending(self):
         super(PaymentTransaction, self)._set_transaction_pending()
 
         for record in self:
-            sales_orders = record.sale_order_ids.filtered(lambda so: so.state == 'draft')
-            sales_orders.force_quotation_send()
-
+            sales_orders = record.sale_order_ids.filtered(lambda so: so.state in ['draft', 'sent'])
             if record.acquirer_id.provider == 'transfer':
                 for so in sales_orders:
                     so.reference = record._compute_sale_order_reference(so)
+            sales_orders.with_context(mark_so_as_sent=True).send_order_confirmation_mail(transaction=record)
```

Comment:
Why is it used here and not in action confirm ?

Reply:
@tde-banana-odoo , As per spec, when the payment is received by portal or by e-commerce then and then we need to send a mail. 
if we write in action confirm then mail will be sent even payment is not received yet and from the backend, someone confirms the order then it will also send mail. Thanks